### PR TITLE
Optimize parser startup time

### PR DESCRIPTION
**Pull request by parandba**
"Optimize parser startup time" issue related code changes.

### CODE REVIEW

File: CLI/src/CLI.csproj [hunk 1]
```diff
@@ -76,7 +76,13 @@
     </ProjectReference>
   </ItemGroup>
   <ItemGroup>
-    <Compile Include="*.cs" />
+    <Compile Include="CLI.cs" />
```
Comment (smedilol) [inline CLI/src/CLI.csproj]: VS try to modify csproj file, but we used to let wildcard *.cs everywhere.
For next Pull request, can you avoid to commit this kind of change, please.

File: CLI/src/Daemon.cs [hunk 2]
```diff
@@ -50,188 +43,194 @@ class Config {
...
-			errmsg += "Try "+PROGNAME+" --help for usage information.";
-			Console.WriteLine(errmsg);
-			return code;
-		}
-
-	}
+            return 0;
+        }
+
+        private static Compiler.DocumentFormat CreateFormat(string encoding, ref Config config)
```
Comment (smedilol) [inline CLI/src/Daemon.cs]: The code of this method is duplicated, can you refactor this?

File: CLI/src/Serialization.cs [hunk 3]
```diff
@@ -6,111 +6,230 @@
...
+            foreach (MsgPack item in msgpack.ForcePathObject("CopyFolders"))
+            {
+                CopyFolders.Add(item.ForcePathObject("CopyFolder").AsString);
+            }
+            config.CopyFolders = CopyFolders;
+
+            List<string> Copies = new List<string>();
+            foreach (MsgPack item in msgpack.ForcePathObject("Copies"))
+            {
+                Copies.Add(item.ForcePathObject("Copies").AsString);
```
Comment (smedilol) [inline CLI/src/Serialization.cs]: It seems there is bug here, as in Encode method, it was "copy"

File: CLI/src/Command.cs [hunk 4]
```diff
@@ -1,83 +1,162 @@
...
+        }
+    }
+
+    /// <summary>
+    /// RunCommandLine class implementing AbstractCommand members.  
+    /// This is called when client passes value "68".
+    /// </summary>
+    internal class RunCommandLine : AbstractCommand
+    {
+        private StringSerializer Deserializer = new StringSerializer();
```
Comment (smedilol) [inline CLI/src/Command.cs]: Field is never used, can you remove it?

File: CLI/src/Serialization.cs [hunk 5]
```diff
@@ -6,111 +6,230 @@
...
+                {
+                    var child = item.AddArrayChild();
+                    child.ForcePathObject("Copy").AsString = copy;
+                }
+            }
+        }
+        internal static Config Decode(MsgPack msgpack)
+        {
+            Config config = new Config();
+            config.ErrorFile = (string)msgpack.ForcePathObject("ErrorFile").AsString;
```
Comment (smedilol) [inline CLI/src/Serialization.cs]: Cast to string is useless here. Can you remove it?

File: CLI/src/Serialization.cs [hunk 6]
```diff
@@ -6,111 +6,230 @@
...
+            config.skeletonPath = (string)msgpack.ForcePathObject("skeletonPath").AsString;
+            config.Codegen = ((string)msgpack.ForcePathObject("Codegen").AsString).Equals("True") ? true : false;
```
Comment (smedilol) [inline CLI/src/Serialization.cs]: End of expression: "? true : false" is useless here. Can you remove it?

File: CLI/src/Daemon.cs [hunk 7]
```diff
@@ -50,188 +43,194 @@ class Config {
...
+                        {
+                            namedPipeClient.Connect(100);
+                        }
+                        catch (TimeoutException tEx)
+                        {
+                            System.Diagnostics.Process process = new System.Diagnostics.Process();
+                            System.Diagnostics.ProcessStartInfo startInfo = new System.Diagnostics.ProcessStartInfo();
+                            startInfo.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+                            startInfo.FileName = "cmd.exe";
+                            startInfo.Arguments = @"/c TypeCobol.CLI.exe";
```
Comment (smedilol) [inline CLI/src/Daemon.cs]: One thing I didn't thought about is that TypeCobol.CLI is not necessarily in the path.
So I see 2 possible solutions:
- Can you retrieve the current folder of the TypeCobol.CLI.exe and then use it to launch this command
- If it doesn't works, then the path to TypeCobol.CLI.exe need to be passed as an argument (with option -K)

Do you see another solution?

File: CLI/src/CLI.csproj (hunk 1 above)
Comment (parandba) [inline CLI/src/CLI.csproj]: ok

File: CLI/src/Daemon.cs (hunk 2 above)
Comment (parandba) [inline CLI/src/Daemon.cs]: Done.

File: CLI/src/Serialization.cs (hunk 5 above)
Comment (parandba) [inline CLI/src/Serialization.cs]: Done.

File: CLI/src/Serialization.cs (hunk 6 above)
Comment (parandba) [inline CLI/src/Serialization.cs]: Config.Codegen accepts bool value.  Hence, if "True" is returned as string, converting it to true as boolean, else false.

File: CLI/src/Serialization.cs (hunk 3 above)
Comment (parandba) [inline CLI/src/Serialization.cs]: Done.

File: CLI/src/Command.cs (hunk 4 above)
Comment (parandba) [inline CLI/src/Command.cs]: Done.

**Comment by smedilol** [comment]
PR merged manually

**Pull request closed without merging**